helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -971,19 +972,24 @@ public void addPreConnectCallback(PreConnectCallback callback) {
 
   @Override
   public boolean isLeader() {
+    return getSessionIdIfLead().isPresent();
+  }
+
+  @Override
+  public Optional<String> getSessionIdIfLead() {
```

Comment:
Still, if you think this is the right name, let us rename isLeader() to isLead(), just make them consistent :).

Reply:
Why we name it `getSessionIfLead()` is because following if should be an adj., eg. `putIfAbsent()`, `computeIfPresent()` etc.. Lead is an adj. Leader is not. So I think from usage, I prefer `getSessionIfLead()`. `isLeader()` is also fine I think.